Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -447,10 +448,11 @@ bool CachedResourceLoader::checkInsecureContent(CachedResource::Type type, const
         // These resource can inject script into the current document (Script,
         // XSL) or exfiltrate the content of the current document (CSS).
         if (auto* frame = this->frame()) {
-            if (!MixedContentChecker::canRunInsecureContent(*frame, m_document->securityOrigin(), url))
+            if (m_document && !MixedContentChecker::canRunInsecureContent(*frame, m_document->securityOrigin(), url))
                 return false;
             auto& top = frame->tree().top();
             auto* localTop = dynamicDowncast<LocalFrame>(top);
+            ASSERT(localTop->document());
             if (&top != frame && localTop && !MixedContentChecker::canRunInsecureContent(*localTop, localTop->document()->securityOrigin(), url))
                 return false;
         }
```

Comment:
Why add a `nullptr` check in one case, but only an `ASSERT()` in the other case?

Also, the `ASSERT(localTop->document())` will crash if `localTop` is `nullptr`.

I'd like to see us be consistent here.

Reply:
My assumption here was that if we are starting from a frame, we would have a document, but since it's a RefPtr I guess there are moments where that may not be true. I'll revise that.